### ISSUE TITLE
Can I only install ns-viewer ? 

### ISSUE THREAD

**Issue by pansanity666** [nerfstudio-project/nerfstudio#1908]
Hi, 
I want to download the ckpt from the remote machine to my laptop, and then use ns-viewer to visualize the ckpt. (For some reason, ssh -L is not work for me.) The question is can I only install ns-viewer in my laptop (Macbook M1) instead of the whole nerfstudio?
Best,

**Comment by brentyi** [comment]
The answer is basically no:
- `ns-viewer` depends on having the whole `nerfstudio` package installed
- `ns-viewer` still depends on CUDA for all of the (practical) models, so unfortunately you won't be able to run it directly on a Macbook

**Comment by pansanity666** [comment]
> The answer is basically no:
> 
> * `ns-viewer` depends on having the whole `nerfstudio` package installed
> * `ns-viewer` still depends on CUDA for all of the (practical) models, so unfortunately you won't be able to run it directly on a Macbook

I found a workaround that is to use IFrame in jupyterlab in the remote machine. However, It shows  renderer disconnected:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/47111102/1c85ea36-fc26-4ebd-9e8b-f70392887d07)

Do you have any idea about it ?

**Comment by brentyi** [comment]
Not really, sorry! Just to confirm are you port forward with `-L`? If you're using Safari, it's also worth trying Chrome.

**Comment by pansanity666** [comment]
> Not really, sorry! Just to confirm are you port forward with `-L`? If you're using Safari, it's also worth trying Chrome.

Do you mean ```ssh -L <port>:localhost:<port> USER@REMOTE.SERVER.IP```?  I actually dont know the remote server ip for some reason.  I have decided to use a more naive solution, ie, to save the rendered image each iteration. LOL

**Comment by brentyi** [comment]
Doesn't seem ideal, but glad you found a solution. 😄 

Just wondering, how are you running the `ns-train` command if you don't have the remote IP? Usually what one would do is SSH in using that command to run the training script.

**Comment by pansanity666** [comment]
> Doesn't seem ideal, but glad you found a solution. 😄
> 
> Just wondering, how are you running the `ns-train` command if you don't have the remote IP? Usually what one would do is SSH in using that command to run the training script.

Actually, the ip address does exist, however, it is not revealed to us users considering the safety issue.  I can only use the remote machine via a specified web page.
